### PR TITLE
Renaming + SPARQL removing

### LINKED ISSUE #9: Pb with some dbpedia queries: StartTag: invalid element name
This query, when limited to a few results (with, e.g. spq_head(10)) works. But running it with a higher limit (for instance 200) causes an error. I suspect it might be due to the encoding of some results but have no clue how to resolve this.

Note that the corresponding SPARQL query, when executed directly in the SPARQL endpoint, runs successfully.

Error : 1: StartTag: invalid element name

  tib=spq_init() %>% 
    spq_prefix(prefixes=c(dbo="http://dbpedia.org/ontology/")) %>% 
    spq_add("?person rdfs:label 'David Bowie'@en")  %>% 
    spq_add("?person ?p ?o") %>%
    spq_head(10) %>% 
    send("https://dbpedia.org/sparql")

### LINKED ISSUE #15: Call "send" "perform" instead?
Also following httr2's vocabulary https://httr2.r-lib.org/reference/req_perform.html

### LINKED ISSUE #19: Remove the {SPARQL} dependency
- Depends on RCurl and XML that are no longer maintained,
- _Depends_ on them in a way that makes glitter have to list SPARQL in Depends vs Imports.

### LINKED ISSUE #26: Notes on the GHA failure on Ubuntu R-devel
It's due to XML installation.

I won't explore as I find #19 more useful (and it'll solve the problem by removing XML from the dependencies tree).

### PR DESCRIPTION
**Pull request by maelle**
Fix #15
Fix #19 
Fix #9
Fix #26 

NOTES
- On parsing: #34
- On the SyMoGih vignette: #35
- On the databnf vignette: #36
- I wonder if the tests should be more specific e.g. adding expectations about tibble columns.

TODOs

- [x] Add tests.

### CODE REVIEW

File: R/build_sparql.R [hunk 1]
```diff
@@ -8,9 +8,9 @@
 #' spq_add("?city wdt:P1082 ?pop") %>%
 #' spq_language("fr") %>%
 #' spq_head(n=5) %>%
-#' build_sparql() %>%
+#' assemble_sparql_query() %>%
```
Comment (maelle) [inline R/build_sparql.R:11]: I found this a better name but I might be wrong, of course.

File: R/send_sparql.R [hunk 2]
```diff
@@ -12,38 +12,57 @@
...
 #''
 #'send_sparql(query=metro_query)
-#' @import SPARQL
 send_sparql=function(query,endpoint="Wikidata"){
   endpoint=tolower(endpoint)
 
   # if endpoint wikidata, use WikidataQueryServiceR::query_wikidata()
   if(endpoint=="wikidata"){
-    tib <- purrr::quietly(WikidataQueryServiceR::query_wikidata)(query)$result
+    return(purrr::quietly(WikidataQueryServiceR::query_wikidata)(query)$result)
```
Comment (maelle) [inline R/send_sparql.R:20]: early return to prevent the if else.

File: R/send_sparql.R [hunk 3]
```diff
@@ -12,38 +12,57 @@
...
+    httr2::req_user_agent("glitter R package (https://github.com/lvaudor/glitter)") %>%
+    httr2::req_retry(max_tries = 3, max_seconds = 120) %>%
+    httr2::req_perform()
+
+    httr2::resp_check_status(resp)
+
+    content = httr2::resp_body_xml(resp)
+    xml2::xml_ns_strip(content)
+
+    # results < result < binding
```
Comment (maelle) [inline R/send_sparql.R:46]: I am not entirely sure yet this is not over simplified. The vignettes did look good.

File: R/send_sparql.R [hunk 4]
```diff
@@ -12,38 +12,57 @@
...
+
   # if endpoint passed as name, get url
-  if(endpoint %in% (usual_endpoints %>%
-                    dplyr::filter(.data$name!="wikidata") %>%
-                    dplyr::pull(.data$name))){
-    url=usual_endpoints %>%
-      dplyr::filter(.data$name == {{ endpoint }}) %>%
-      dplyr::select(.data$url) %>%
-      dplyr::pull()
-    tib <- SPARQL::SPARQL(url=url,
```
Comment (maelle) [inline R/send_sparql.R:32]: In practice this uses the default `format` value for `SPARQL::SPARQL()` which is "xml", but this gets a data.frame. :raised_eyebrow: https://github.com/cran/SPARQL/blob/51b993f16cb924f2ccc325f08f42da3420ea08bc/R/SPARQL.R#L63

Comment (maelle) [reply R/send_sparql.R:32]: That argument was more for the HTTP header than for the output I suppose.

Comment (maelle) [reply R/send_sparql.R:32]: Found my favorite part of parsing in {SPARQL}: https://github.com/cran/SPARQL/blob/51b993f16cb924f2ccc325f08f42da3420ea08bc/R/SPARQL.R#L141 :grin:

File: R/send_sparql.R [hunk 5]
```diff
@@ -12,38 +12,70 @@
...
+    content = httr2::resp_body_json(resp)
+
+    # Adapted from https://github.com/wikimedia/WikidataQueryServiceR/blob/accff89a06ad4ac4af1bef369f589175c92837b6/R/query.R#L56
+    if (length(content$results$bindings) > 0) {
+        data_frame <- purrr::map_dfr(content$results$bindings, function(binding) {
+          return(purrr::map_chr(binding, ~ .x$value))
+        })
+        datetime_columns <- purrr::map_lgl(content$results$bindings[[1]], function(binding) {
+          if ("type" %in% names(binding)) {
+            return(binding[["type"]] == "http://www.w3.org/2001/XMLSchema#dateTime")
```
Comment (maelle) [inline R/send_sparql.R:57]: to be tested with queries where we know there will be a datatime, and to be generalized to other special types.

File: tests/testthat/test-send_sparql.R [hunk 6]
```diff
@@ -1,4 +1,5 @@
 test_that("send_sparql() returns tibble", {
+  skip_if_offline()
```
Comment (maelle) [inline tests/testthat/test-send_sparql.R:2]: WikidataQueryServiceR uses httr so I'd have to use httptest, not httptest2 for mocking.

File: R/build_sparql.R (hunk 1 above)
Comment (lvaudor) [reply R/build_sparql.R:11]: it's fine but maybe a bit long (sorry I comment whereas you're already a bit far in the process of renaming and changing this everywhere...). Would spq_build (and spq_process) sound fine to you?

File: R/send_sparql.R (hunk 2 above)
Comment (lvaudor) [reply R/send_sparql.R:20]: yes

File: R/build_sparql.R (hunk 1 above)
Comment (maelle) [reply R/build_sparql.R:11]: or spq_assemble? I can use your name suggestion next week, easy enough to change, no worries.

Comment (maelle) [reply R/build_sparql.R:11]: Ou alors spq_query mais y'a un risque de confondre avec le verbe "query"

**Pull request merged**